File: crates/vaultic-types/src/vault.rs
//! Vault item and folder types.

use chrono::{DateTime, Utc};
use serde::{Deserialize, Serialize};
use uuid::Uuid;

#[derive(Debug, Clone, Serialize, Deserialize, PartialEq)]
#[serde(rename_all = "snake_case")]
pub enum ItemType {
    Login,
    SecureNote,
    Card,
    Identity,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct VaultItem {
    pub id: Uuid,
    pub user_id: Uuid,
    pub folder_id: Option<Uuid>,
    pub item_type: ItemType,
    /// Base64-encoded nonce + ciphertext
    pub encrypted_data: String,
    pub device_id: String,
    pub version: i32,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
    pub deleted_at: Option<DateTime<Utc>>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
pub struct Folder {
    pub id: Uuid,
    pub user_id: Uuid,
    pub encrypted_name: String,
    pub parent_id: Option<Uuid>,
    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

File: shared/types/src/vault.ts
// Vault item and folder types

export enum ItemType {
  Login = 'login',
  SecureNote = 'secure_note',
  Card = 'card',
  Identity = 'identity',
}

export interface VaultItem {
  id: string;
  folder_id?: string;
  item_type: ItemType;
  /** Base64-encoded nonce + ciphertext */
  encrypted_data: string;
  device_id: string;
  version: number;
  created_at: string;
  updated_at: string;
  deleted_at?: string;
}

export interface Folder {
  id: string;
  encrypted_name: string;
  parent_id?: string;
  created_at: string;
  updated_at: string;
}

/** Decrypted login credential for UI display */
export interface LoginCredential {
  name: string;
  url?: string;
  username: string;
  password: string;
  notes?: string;
}
